### ISSUE TITLE
Cannot create a Down payment invoice with sales user

### ISSUE THREAD

**Issue by MarkingSC** [odoo/odoo#46014]
Impacted versions:
v13
Steps to reproduce:
Create a sale order, confirm it, click on _Create Invoice_, choose down payment (any one), then confirm invoice creation.

Current behavior:
Odoo is sending a permission error, that doesn't happen when creating a regular invoice

Expected behavior:
If _Create invoice_ is there for sales users, and also the down payment options are, this kind of user should be able to create a down payment invoice. 

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/26973424/75127388-79f51c80-5684-11ea-848b-02dbdd97e0b4.png)

I don't know if running "create_invoices" method from sale.advance.payment.inv  as Sudo would be the solution. Even I tried it but im getting a permission error on

**Comment by 0yik** [comment]
@MarkingSC sales user fail when create product.product model. You can add new ACL, give sales users groups to create product.product model

**Comment by MarkingSC** [comment]
@0yik I did it, but then it fails on permission creating product.template records...

**Comment by 0yik** [comment]
@MarkingSC So, you need product.template too

**Comment by MarkingSC** [comment]
@0yik then I will get the error on writing account.move, but that's what I want to avoid. Sales users should not write on invoice records, they should only create the record but not write on it.

**Comment by pedrobaeza** [comment]
It seems you have your system misconfigured, as this shouldn't happen on a proper installation. Are you able to reproduce the problem in runbot infrastructure (http://runbot.odoo.com/runbot - admin/admin)?

If so, please explain with all the possible detailed steps to get to the error. Not having them, it's very difficult to hunt down the problem. Please don't be shy of adding details, and even better if you provide some screenshots or videos.

If not, this can be due to a configuration change, a custom/community module you have deployed in your DB or an outdated code. Please report back if it's the case.

For keeping a clean house in this issue tracker, if you don't answer in several days, this will be closed assuming it was one of the stated things.

**Comment by MarkingSC** [comment]
Hello @pedrobaeza ,  I reproduced this error on runbot (http://1604751-13-0-e3f805.runbot16.odoo.com/), I set a Salesman user with the groups set as below:
![image](https://user-images.githubusercontent.com/26973424/75808629-05d60b00-5d4d-11ea-9bea-8865c7d7a7e3.png)

It supposed the user to be a salesman only, when I login with demo/demo I did these steps:

1. I went to Sales.
2. I selected a sale order.
3. I clicked on _Create _invoice__ .
4. I selected Down payment (percentage) option and set 50%.
5. When clicking on _Create and view _invoice__ or _Create invoice_ it throws this error:
![image](https://user-images.githubusercontent.com/26973424/75809085-c9ef7580-5d4d-11ea-8f9f-ccaac0091048.png)

**Comment by pedrobaeza** [comment]
But this seems that this is because the down payment product has been removed.

**Comment by MarkingSC** [comment]
Yeah, It looks like that, but I didn't do anything else than setting the user groups for demo user. 

... But, even when Down payment product has been created from creating a Down Payment invoice with Admin user, then I tried to do it with Demo user again and throws this error:
![image](https://user-images.githubusercontent.com/26973424/75810322-fdcb9a80-5d4f-11ea-8d27-a72b3b4442dc.png)

With all of those errors, I suppose Down Payment invoices can be created only by an Accounting user... but I'm not sure if it should be like that, considering that Odoo lets Sales users to create a regular invoice but not a Down Payment invoice.

**Comment by pedrobaeza** [comment]
Well, that's weird also, as that only happens if you validate the invoice and it contains analytic account in it. Are you sure you are not doing anything more or any custom thing?

**Comment by MarkingSC** [comment]
I'm sure, I just configured the Sales user, as you can see on the comments above...

For my business I bypassed the function using with_user method, it creates the invoice from root user, It's kind of hacky, but I got no option...
![image](https://user-images.githubusercontent.com/26973424/75813305-5d787480-5d55-11ea-9f84-ce22e740f95c.png)

**Comment by pedrobaeza** [comment]
@dbo-odoo are you aware of this issue?

**Comment by pedrobaeza** [comment]
@dbo-odoo can you please check this?

**Comment by Yenthe666** [comment]
@dbo-odoo friendly reminder.
